### PR TITLE
Overhaul look around panels

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Interface "Overhaul look around panels"

#### Purpose of change
Shows more information in look around menus and makes it easier to read at a glance.

#### Describe the solution
Hijack code from #36324.

#### Testing
Panels render as expected.

#### Additional context
This is the same as where #36324 left off with some minor code changes. Still need to make it format right for translations.

### CODE REVIEW

**Comment by esotericist** [comment]
As I said in the previous PR, I'd like this to be gated by an option because a lot of the choices here are deleterious to how I process information.

long term, what we want is for this sort of thing to be json-driven, but that's a much bigger project.

**Comment by ymber** [comment]
This should be alright for translation formatting now. Some changes also to improve UX.

**Comment by Night-Pryanik** [comment]
@ymber could you please provide some screenshots with your rework?

**Comment by ymber** [comment]
Everywhere that has the Label: Value pattern the second string is printed one space after the translated label so it should be adaptive enough.

**Comment by ymber** [comment]
Some more complex panels with a load of stuff on the same tile. I think the last one is showing bugs with the trap information not ending with a newline and the health bar for the zombie not showing at extremely low health.

**Comment by Night-Pryanik** [comment]
Cool, thanks. LGTM.

**Comment by captnblood** [comment]
for the record, i think this is completely missing the whole point of my original PR.
what you did here makes as much sense as adopting a cat and try to force him to bark.
my PR was bringing order, better presentation, what you did is reverting it back to the same kind of inordered mess as found currently in the "game" (i consider it more like a giant, obese bug masquarading as a game and getting worse every release.. just to be clear on that).

**Comment by ZhilkinSerg** [comment]
> for the record, i think this is completely missing the whole point of my original PR.
> what you did here makes as much sense as adopting a cat and try to force him to bark.
> my PR was bringing order, better presentation, what you did is reverting it back to the same kind of inordered mess as found currently in the "game" (i consider it more like a giant, obese bug masquarading as a game and getting worse every release.. just to be clear on that).

I am kindly asking you to refrain from posting unfair insults to work of the other people.

**Comment by captnblood** [comment]
@ZhilkinSerg 
there is no insults. i simply stated my opinion on this PR which took inspiration and code from mine. i also stated my opinion on this "game". these opinions are not "unfair" and are certainly merited. i deleted my repo and my PRs for that reason. now do what you wish i'm already out.

**Comment by wapcaplet** [comment]
I had trouble seeing the differences from the current version to this one, so I took a couple screenshots with a monster and NPC. On the left is the master branch as of 889f65752, and on the right is the new panel as of 337b496debc

Monster:
image

NPC:
image

Overall the changes look fairly subtle. I like the white highlight for the terrain (especially since it is listed at the top), and the red highlight for monster awareness. The addition of ahealth bar is definitely of the good, and the blue colors for NPC items was kind of distracting, so grey text makes sense there.

@ymber Could the "Lighting:" label be restored, instead of just having the word "bright" (or "dark" etc.)? This would be more consistent with the new labels. Also, the pink color-coding for the NPC was a useful highlight to distinguish them from monsters - would you consider restoring as well?

My only other concern is the increased vertical space requirement. NPC attitude ("Neutral") could perhaps go on the same line with "Aware of your presence" to save a line.

**Comment by esotericist** [comment]
So, while we're tampering with this, I'd like to bring up something that came up in discord recently
specifically: "Aware of your presence/Knows you are there"

this line is actually a complete lie. The logic behind it is not actually keying off of whether the critter/character is aware of you, but whether or not the critter/character has vision capability to the tile you're in. Notably, when `X` peeking, it can show "Aware of your presence" because you've moved to a tile they have line of sight to, even if they can't actually see *you*.

We might want to pursue phrasing along the lines of: "Can see to your current location". I also agree with wapcaplet about being interested in combining it with the hostility indicator, although that's tricky with longer verbiage.

Another thing we might consider adding at the bottom: terrain/furniture descriptions. We *have* these in-game, but they are exceedingly unobvious to get at so most people don't know they exist (which contributes to many of them being incorrect, since they're never seen). They're comparatively unimportant, so shoving them at the bottom keeps them from eating space that belongs to higher priority things.

I have other desires, but I think they fall into "highly subjective bits, best reserved for hypothetical customization". I think aside from what's been mentioned above, this should be fine.

**Comment by esotericist** [comment]
Oh, one thing:
Can you pad the critter health with `.` so it has a consistent width?

**Comment by ymber** [comment]
Restored lighting label and reworded the awareness indicator. When AI map awareness gets reworked at some point in the future we should revisit how we communicate monster awareness to the player. I added dot padding to the health bars in this panel and in the shift+v panel while I was on.

I was considering making monster and NPC names the same color as their symbols. This is the way the shift+v menu went and it seems like the best solution if white names for all monsters and NPCs makes them harder to distinguish at a glance.

I think putting furniture and terrain descriptions under everything else would be bad for how information is clustered. Given how much screen space they'd take the way I'd go for that would be having an extended descriptions key in the panel that shows all that. If that's added we might want to hide monster descriptions behind it as well.

**Comment by Zireael07** [comment]
> I was considering making monster and NPC names the same color as their symbols.

YES PLEASE.

File: src/game.cpp [hunk 1]
```diff
@@ -5866,70 +5866,72 @@ void game::print_visibility_info( const catacurses::window &w_look, int column,
...
-                        u.has_trait( trait_ILLITERATE ) ? _( "Sign: ???" ) : _( "Sign: %s" ), signage );
+        std::string sign_string = u.has_trait( trait_ILLITERATE ) ? "???" : signage;
+        mvwprintz( w_look, point( column, ++line ), c_light_gray, _( "Sign: %s" ), sign_string );
     }
 
+    // Print light level on the selected tile.
+    std::pair<std::string, nc_color> ll = get_light_level( std::max( 1.0,
+                                          LIGHT_AMBIENT_LIT - m.ambient_light_at( lp ) + 1.0 ) );
+    mvwprintz( w_look, point( column, ++line ), c_light_gray, _( "Lighting: " ) );
+    mvwprintz( w_look, point( column + utf8_width( _( "Lighting: " ) ), line ), ll.second, ll.first );
```
Comment (Night-Pryanik) [inline src/game.cpp:5915]: Seems like there should be `Lighting: foo` line, but according to your screenshots there are only `foo` text.
изображение

File: src/game.cpp [hunk 2]
```diff
@@ -5866,70 +5866,72 @@ void game::print_visibility_info( const catacurses::window &w_look, int column,
...
     if( m.has_furn( lp ) ) {
-        tile += "; " + m.furnname( lp );
+        mvwprintz( w_look, point( column, ++line ), c_light_blue, m.furnname( lp ) );
     }
 
+    // Cover percentage from terrain and furniture next.
+    fold_and_print( w_look, point( column, ++line ), max_width, c_light_gray, _( "Cover: %d%%" ),
+                    m.coverage( lp ) );
+    // Terrain and furniture flags next.
+    mvwprintz( w_look, point( column, ++line ), c_light_gray, "%s", m.features( lp ) );
```
Comment (Night-Pryanik) [inline src/game.cpp:5893]: Could you please check for this line overlapping to the next line if there are too many map features here to fit in one line, like in this pre-your-changes screenshot:
изображение

File: src/game.cpp (hunk 1 above)
Comment (ymber) [reply src/game.cpp:5920]: That screenshot is from before I added the `Lighting: ` string. It's there in the current code.

**Comment by ymber** [comment]
Fixed terrain and furniture flags disappearing across lines.

**Comment by ZhilkinSerg** [comment]
```
astyle regressions found.
Formatted  src/game.cpp Formatted  src/npc.cpp
```

File: src/game.cpp [hunk 3]
```diff
@@ -6145,7 +6164,7 @@ void game::print_graffiti_info( const tripoint &lp, const catacurses::window &w_
     const int max_width = getmaxx( w_look ) - column - 2;
     if( m.has_graffiti_at( lp ) ) {
         fold_and_print( w_look, point( column, ++line ), max_width, c_light_gray,
-                        m.ter( lp ) == t_grave_new ? _( "Graffiti: %s" ) : _( "Inscription: %s" ),
+                        m.ter( lp ) == t_grave_new ? _( "Graffiti: %s" ) : _( "inscription: %s" ),
```
Comment (ZhilkinSerg) [inline src/game.cpp:6167]:
```suggestion
                        m.ter( lp ) == t_grave_new ? _( "Graffiti: %s" ) : _( "Inscription: %s" ),
```

File: src/game.cpp [hunk 4]
```diff
@@ -6123,12 +6142,12 @@ void game::print_items_info( const tripoint &lp, const catacurses::window &w_loo
...
-            if( it->second > 1 ) {
-                trim_and_print( w_look, point( column, ++line ), max_width, c_white,
+            if( it.second.first > 1 ) {
+                trim_and_print( w_look, point( column, ++line ), max_width, it.second.second,
                                 pgettext( "%s is the name of the item.  %d is the quantity of that item.", "%s [%d]" ),
-                                it->first.c_str(), it->second );
+                                it.first.c_str(), it.second.first );
             } else {
-                trim_and_print( w_look, point( column, ++line ), max_width, c_white, it->first );
+                trim_and_print( w_look, point( column, ++line ), max_width, it.second.second, it.first );
```
Comment (ZhilkinSerg) [inline src/game.cpp:6150]:
```suggestion
                trim_and_print( w_look, point( column, ++line ), max_width, it->second.second, it->first );
```

File: src/game.cpp [hunk 5]
```diff
@@ -6123,12 +6142,12 @@ void game::print_items_info( const tripoint &lp, const catacurses::window &w_loo
                 break;
             }
 
-            if( it->second > 1 ) {
-                trim_and_print( w_look, point( column, ++line ), max_width, c_white,
+            if( it.second.first > 1 ) {
+                trim_and_print( w_look, point( column, ++line ), max_width, it.second.second,
                                 pgettext( "%s is the name of the item.  %d is the quantity of that item.", "%s [%d]" ),
-                                it->first.c_str(), it->second );
+                                it.first.c_str(), it.second.first );
```
Comment (ZhilkinSerg) [inline src/game.cpp:6148]:
```suggestion
                                it->first.c_str(), it->second.first );
```

File: src/game.cpp [hunk 6]
```diff
@@ -6123,12 +6142,12 @@ void game::print_items_info( const tripoint &lp, const catacurses::window &w_loo
                 break;
             }
 
-            if( it->second > 1 ) {
-                trim_and_print( w_look, point( column, ++line ), max_width, c_white,
+            if( it.second.first > 1 ) {
+                trim_and_print( w_look, point( column, ++line ), max_width, it.second.second,
```
Comment (ZhilkinSerg) [inline src/game.cpp:6146]:
```suggestion
                trim_and_print( w_look, point( column, ++line ), max_width, it->second.second,
```

File: src/game.cpp [hunk 7]
```diff
@@ -6123,12 +6142,12 @@ void game::print_items_info( const tripoint &lp, const catacurses::window &w_loo
                 break;
             }
 
-            if( it->second > 1 ) {
-                trim_and_print( w_look, point( column, ++line ), max_width, c_white,
+            if( it.second.first > 1 ) {
```
Comment (ZhilkinSerg) [inline src/game.cpp:6145]:
```suggestion
            if( it->second.first > 1 ) {
```

**Comment by ZhilkinSerg** [comment]
Caught one thing with graffiti (not caused by current PR) - https://github.com/CleverRaven/Cataclysm-DDA/pull/30281/files#r429820719. Otherwise looks good for me.

**Pull request merged**